### PR TITLE
4 - Change refrence from tiniyiiod to iio_desc

### PR DESCRIPTION
**Pull request by mchindri**
Only last commit is relevant.
will be rebased when https://github.com/analogdevicesinc/no-OS/pull/657 after is merged

### CODE REVIEW

File: iio/iio_app/iio_app.c [hunk 1]
```diff
@@ -106,7 +106,7 @@ int32_t iio_app(struct iio_app_desc *desc)
 	int32_t status;
 
 	while(1) {
-		status = tinyiiod_read_command(desc->iiod);
+		status = iio_step(desc->iio);
```
Comment (cristiantpop) [inline iio/iio_app/iio_app.c:109]: This can be renamed to `iio_read_command`, maybe.

Comment (mchindri) [reply iio/iio_app/iio_app.c:109]: I am not sure how to name it.  iio_read_command makes me think that I should expect some output from the function, that's way I though it is not a good idea to name it this way.

**Comment by mchindri** [comment]
v2: 
 - Rebase onto master

**Comment by mchindri** [comment]
v3:
 - Rebase onto https://github.com/analogdevicesinc/no-OS/pull/663

**Comment by mchindri** [comment]
v3:
 - Rebase onto https://github.com/analogdevicesinc/no-OS/pull/663

**Comment by mchindri** [comment]
v4:
 -rebase onto master

**Comment by mchindri** [comment]
v5:
 - fix astyle

**Pull request merged**